src/backend/dx12/src/device.rs

```diff
@@ -1143,31 +1242,7 @@ impl d::Device<B> for Device {
     }
 
     fn create_shader_module(&self, raw_data: &[u8]) -> Result<n::ShaderModule, d::ShaderError> {
-        let mut ast = Self::parse_spirv(raw_data)?;
-        if ast.get_specialization_constants().map_err(gen_query_error)?.len() > 0 {
-            return Ok(n::ShaderModule::Spirv(raw_data.into()))
-        }
-
-        Self::patch_spirv_resources(&mut ast)?;
-
-        let shader_model = hlsl::ShaderModel::V5_1;
-        let shader_code = Self::translate_spirv(&mut ast, shader_model)?;
-        debug!("SPIRV-Cross generated shader:\n{}", shader_code);
-
-        let mut shader_map = BTreeMap::new();
-        let entry_points = ast.get_entry_points().map_err(gen_query_error)?;
-        for entry_point in entry_points {
-            let stage = conv::map_execution_model(entry_point.execution_model);
-            let shader_blob = Self::compile_shader(
-                stage,
-                shader_model,
-                &entry_point.name,
-                shader_code.as_bytes(),
-            )?;
-
-            shader_map.insert(entry_point.name, shader_blob);
-        }
-        Ok(n::ShaderModule::Compiled(shader_map))
+        Ok(n::ShaderModule::Spirv(raw_data.into()))
```

Comment:
can we detect here if any push constants are present and go to the `Compiled` fast path?

Reply:
Hm requires two things:
- Checking if it's allowed to not declare constants buffers from the root signature in the shaders
- Adjust register spaces of descriptor tables to enforce that space0 will always be reserved for root constants even if the pipeline layout doesn't have push constants.